app/workers/send_user_invitation_worker.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class SendUserInvitationWorker < ApplicationJob
+  queue_as :default
+
+  RETRY_ERRORS = [
+    OpenSSL::SSL::SSLError,
+    Net::SMTPAuthenticationError,
+    Net::SMTPFatalError,
+    Net::SMTPServerBusy,
+    Net::SMTPSyntaxError,
+    Net::SMTPUnknownError,
+    Net::SMTPUnsupportedCommand,
+    SocketError,
+  ].freeze
+
+  DISCARD_ERRORS = [
+    ActiveRecord::RecordNotFound
+  ].freeze
+
+  def perform(invitation_id)
+    invitation = Invitation.find(invitation_id)
+
+    mailer = invitation.account.provider? ? ProviderInvitationMailer : InvitationMailer
+    mailer.invitation(invitation).deliver_now!
+
+    invitation.update!(sent_at: Time.zone.now)
+  rescue *RETRY_ERRORS => error
+    logger.error(error.message)
+  rescue *DISCARD_ERRORS => error
+    logger.error(error.message)
+    false
```

Comment:
Rescuing here is ok for me, but better doing `rescue_from` IMO.
Example:
https://github.com/3scale/porta/blob/765e5f013bc30bb0b152294fce81e737660de6ad/app/workers/delete_plain_object_worker.rb#L19-L22

Reply:
I've been trying to use `rescue_from` but can't use the splat operator.. I'll stick with `rescue` for now